scripts/azure_logs_integration.js

```diff
@@ -0,0 +1,699 @@
+var https = require("https");
+
+const VERSION = "1.1.0";
+
+const STRING = "string"; // example: 'some message'
+const STRING_ARRAY = "string-array"; // example: ['one message', 'two message', ...]
+const JSON_OBJECT = "json-object"; // example: {"key": "value"}
+const JSON_ARRAY = "json-array"; // example: [{"key": "value"}, {"key": "value"}, ...] or [{"records": [{}, {}, ...]}, {"records": [{}, {}, ...]}, ...]
+const BUFFER_ARRAY = "buffer-array"; // example: [<Buffer obj>, <Buffer obj>]
+const JSON_STRING = "json-string"; // example: '{"key": "value"}'
+const JSON_STRING_ARRAY = "json-string-array"; // example: ['{"records": [{}, {}]}'] or ['{"key": "value"}']
+const INVALID = "invalid";
+const JSON_TYPE = "json";
+const STRING_TYPE = "string";
+
+const KF_HTTP_PORT = process.env.KF_PORT || 443;
+const KF_REQUEST_TIMEOUT_MS = 10000;
+const KF_TAGS = process.env.KF_TAGS || ""; // Replace '' by your comma-separated list of tags
+const KF_SERVICE = process.env.KF_SERVICE || "azure";
+const KF_SOURCE = process.env.KF_SOURCE || "azure";
+const KF_SOURCE_CATEGORY = process.env.KF_SOURCE_CATEGORY || "azure";
+const KF_PARSE_DEFENDER_LOGS = process.env.KF_PARSE_DEFENDER_LOGS; // Boolean whether to enable special parsing of Defender for Cloud logs. Set to 'false' to disable
+
+const MAX_RETRIES = 4; // max number of times to retry a single http request
+const RETRY_INTERVAL = 250; // amount of time (milliseconds) to wait before retrying request, doubles after every retry
+
+// constants relating to Defender for Cloud logs
+const MSFT_DEFENDER_FOR_CLOUD = "Microsoft Defender for Cloud";
+const AZURE_SECURITY_CENTER = "Azure Security Center";
+const SECURITY_ASSESSMENTS = "Microsoft.Security/assessments";
+const SECURITY_SUB_ASSESSMENTS =
+  "Microsoft.Security/assessments/subAssessments";
+const SECURITY_COMPLIANCE_ASSESSMENTS =
+  "Microsoft.Security/regulatoryComplianceStandards/regulatoryComplianceControls/regulatoryComplianceAssessments";
+const SECURITY_SCORES = "Microsoft.Security/secureScores";
+const SECURITY_SCORE_CONTROLS =
+  "Microsoft.Security/secureScores/secureScoreControls";
+const DEFENDER_FOR_CLOUD_PRODUCTS = [
+  MSFT_DEFENDER_FOR_CLOUD,
+  AZURE_SECURITY_CENTER,
+];
+const DEFENDER_FOR_CLOUD_RESOURCE_TYPES = [
+  SECURITY_ASSESSMENTS,
+  SECURITY_SUB_ASSESSMENTS,
+  SECURITY_COMPLIANCE_ASSESSMENTS,
+  SECURITY_SCORES,
+  SECURITY_SCORE_CONTROLS,
+];
+
+/*
+To scrub PII from your logs, uncomment the applicable configs below. If you'd like to scrub more than just
+emails and IP addresses, add your own config to this map in the format
+NAME: {pattern: <regex_pattern>, replacement: <string to replace matching text with>}
+*/
+const SCRUBBER_RULE_CONFIGS = {
+  // REDACT_IP: {
```

Comment:
Why are these commented and also the constant below?

If not needed, should we just remove them?

Reply:
The scrubber part is to redact the sensitive information (PII) in the logs but not being used currently. So, it can be removed for now if we don't have to redact PII